src/ralph/backends/storage/fs.py

```diff
@@ -0,0 +1,130 @@
+"""FileSystem storage backend for Ralph"""
+
+import datetime
+import logging
+import os
+import sys
+
+from ..mixins import HistoryMixin
+from .base import BaseStorage
+
+logger = logging.getLogger(__name__)
+
+
+class FSStorage(HistoryMixin, BaseStorage):
+    """FileSystem storage backend"""
+
+    name = "fs"
+
+    _fspath = "fs_archives/"
+
+    def __init__(self):
+        if not os.path.isdir(self._fspath) and not os.path.exists(self._fspath):
+            logger.debug(
+                "No existing FileSystemStorage, creating a new one at %s", self._fspath
+            )
+            os.mkdir(self._fspath)
+        logger.debug("FileSystemStorage's path: %s", self._fspath)
+
+    def _get_filepath(self, name):
+        """Return path of the archive in the fsstorage, or throws an exception if not found"""
+        path = os.path.join(self._fspath, name)
+        if not os.path.isfile(path):
+            logger.error("%s does not exist", path)
+            raise FileNotFoundError()
+        return path
+
+    def _details(self, name):
+        """Get name archive details"""
+        path = self._get_filepath(name)
+        details = {}
+        details["filename"] = name
+        details["size"] = os.path.getsize(path)
+
+        return details
+
+    def list(self, details=False, new=False):
+        """List files in the storage backend"""
+        archives = os.listdir(self._fspath)
+        logger.debug("Found %d archives", len(archives))
+
+        if new:
+            archives = set(archives) - set(
+                entry.get("id")
+                for entry in filter(
+                    lambda e: e["backend"] == self.name and e["command"] == "fetch",
+                    self.history,
+                )
+            )
+            logger.debug("New archives: %d", len(archives))
+
+        for archive in archives:
+            yield self._details(archive) if details else archive
+
+    def url(self, name):
+        """Get `name` file absolute URL"""
+
+        path = self._get_filepath(name)
+        return path
+
+    def read(self, name, chunk_size=4096):
+        """Read `name` file and stream its content by chunks of a given size"""
+
+        logger.debug("Getting archive: %s", name)
+
+        path = self._get_filepath(name)
+        file = open(path, "rb")
+        chunk = file.read(chunk_size)
+        while chunk:
+            sys.stdout.buffer.write(chunk)
+            chunk = file.read(chunk_size)
+        file.close()
```

Comment:
Use the `with` statement while opening files and also use the new walrus operator when you can!

```suggestion
        with open(path, "rb") as file:
            while chunk := file.read(chunk_size):
                sys.stdout.buffer.write(chunk)
```